File: src/MathShortcuts.ts
const Tau = 2 * Math.PI;
let abs = (n: number) => Math.abs(n);
let floor = (n: number) => Math.floor(n);
let sqrt = (n: number) => Math.sqrt(n);
let cos = (n: number) => Math.cos(n);
let sin = (n: number) => Math.sin(n);
let pow = (n: number, e: number) => Math.pow(n, e);
let log = (n: number) => Math.log(n);
let max = (...n: number[]) => Math.max(...n);
let min = (...n: number[]) => Math.min(...n);
let round = (n: number) => Math.round(n);
let ceil = (n: number) => Math.ceil(n);

let clamp = (n: number, min: number, max: number) => 
    n < min 
        ? min
        : n > max 
            ? max   
            : n;

export {
    abs,
    floor,
    ceil,
    sqrt,

    cos,
    sin,

    pow,
    log, 

    round,
    clamp,
    min,
    max,
    
    Tau
}

File: lua/src/MathShortcuts.lua
local ____exports = {}
local Tau = 2 * math.pi
local function abs(n)
    return math.abs(n)
end
local function floor(n)
    return math.floor(n)
end
local function sqrt(n)
    return math.sqrt(n)
end
local function cos(n)
    return math.cos(n)
end
local function sin(n)
    return math.sin(n)
end
local function pow(n, e)
    return n ^ e
end
local function log(n)
    return math.log(n)
end
local function max(...)
    return math.max(...)
end
local function min(...)
    return math.min(...)
end
local function round(n)
    return math.floor(n + 0.5)
end
local function ceil(n)
    return math.ceil(n)
end
local function clamp(n, min, max)
    return n < min and min or (n > max and max or n)
end
____exports.abs = abs
____exports.floor = floor
____exports.ceil = ceil
____exports.sqrt = sqrt
____exports.cos = cos
____exports.sin = sin
____exports.pow = pow
____exports.log = log
____exports.round = round
____exports.clamp = clamp
____exports.min = min
____exports.max = max
____exports.Tau = Tau
return ____exports
